src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -74,7 +75,15 @@ uiRoutes
   }
 });
 
-app.controller('discover', function ($scope, config, courier, $route, $window, Notifier,
+app
+  .directive('discoverApp', function () {
+    return {
+      controllerAs: 'discoverApp',
```

Comment:
Minor suggestion: if we add `restrict: 'E',` then we can make the markup a bit cleaner (below).